instrumentation/jdbc/library/src/test/groovy/io/opentelemetry/instrumentation/jdbc/OpenTelemetryDriverTest.groovy

```diff
@@ -121,6 +121,19 @@ class OpenTelemetryDriverTest extends Specification {
     connection == null
   }
 
+  def "verify registration of local driver"() {
+    when:
+    deregisterTestDriver()
+    TestDriver driver = new TestDriver()
+    OpenTelemetryDriver.INSTANCE.registerDriver(driver)
+    def connection = OpenTelemetryDriver.INSTANCE.connect("jdbc:otel:test:", null)
+    OpenTelemetryDriver.INSTANCE.deregisterDriver(driver)
+
+    then:
+    connection != null
+    connection instanceof OpenTelemetryConnection
+  }
+
```

Comment:
I think we need more test cases:
* driver coming from DriverManager looses
* Not calling registerDriver(driver) but having  DriverManager will work (legacy behaviour)
* what will happen if we add 2 clashing drivers?

Reply:
Added new tests, legacy behavior is covered by existing tests, I ensured it is not affected by a new ones.

Reply:
Should look better now, thank you for your time.